python/lsst/summit/utils/astrometry/anet.py

```diff
@@ -0,0 +1,428 @@
+# This file is part of summit_utils.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import os
+import shutil
+import subprocess
+import tempfile
+import numpy as np
+from astropy.io import fits
+import time
+import uuid
+import warnings
+
+import lsst.geom as geom
+
+from .utils import headerToWcs
+
+__all__ = ['AstrometryNetResult', 'CommandLineSolver', 'OnlineSolver']
+
+
+class AstrometryNetResult():
+    """Minimal wrapper class to construct and return results from the command
+    line fitter.
+
+    Constructs a DM wcs from the output of the command line fitter, and sets
+    the plate scale on the class for use in the scatter measurement. This loads
+    the corr files from the fit and calculates the scatter in pixels and
+    arcseconds.
+
+    Parameters
+    ----------
+    wcsFile : `str`
+        The path to the .wcs file from the fit.
+    corrFile : `str`
+        The path to the .corr file from the fit.
+    """
+    wcs = None
+    scatterPixels = None
+    scatterArcseconds = None
+    plateScale = None
+
+    def __init__(self, wcsFile, corrFile=None):
+        self.wcsFile = wcsFile
+        self._buildWcs()  # create the DM wcs from the a.net wcs file. Sets self.plateScale from fitted wcs
+
+        self.corrFile = corrFile
+        self._calcScatter()  # sets self.scatterPixels and self.scatterArcseconds
+
+    def _buildWcs(self):
```

Comment:
Do you actually want this object to be mutable? I'm guessing not since you made buildWcs and calcScatter single-underscore and never call them outside init.

In that case, I suggest making this a frozen dataclass (`from dataclasses import dataclass` and add `@dataclass(frozen=True)`). Then, replace the existing class attributes with methods using `@cached_property` (from functools). The nice thing about that is that the wcs method will contain all but the last line of what is now `_buildWcs`, whereas `plateScale` will be a one-liner calling `self.wcs.getPixelScale().asArcseconds()`, and you won't need any single-underscore methods at all. Plus, if you really care about performance, if you never call those methods it won't bother to read the file and build the wcs in the first place.